bucket/ddc_bucket/cluster/messages.rs

```diff
@@ -61,9 +61,9 @@ impl DdcBucket {
         let manager = Self::env().caller();
         let mut nodes = Vec::<(NodeId, &Node)>::new();
 
-        for node_id in node_ids {
-            let node = self.nodes.get(node_id)?;
-            nodes.push((node_id, node));
+        for node_id in &node_ids {
```

Comment:
for node_id in node_ids.iter() {
            let node = self.nodes.get(*node_id)?;
            nodes.push((*node_id, node));

Reply:
You should run `cargo +nightly clippy`